test/dict.jl

```diff
@@ -388,6 +388,11 @@ d = Dict('a'=>1, 'b'=>1, 'c'=> 3)
 @test [d[k] for k in keys(d)] == [d[k] for k in eachindex(d)] ==
       [v for (k, v) in d] == [d[x[1]] for (i, x) in enumerate(d)]
 
+# generators, similar
+d = Dict(:a=>"a")
+@test @inferred(map(identity, d)) == d
+@test @inferred(map(p->p.first=>p.second[1], d)) == Dict(:a=>'a')
+@test @inferred(map(p->p.second, d)) == Any["a"]
```

Comment:
In theory it could be a deprecation, but the warning should not depend on
type inference, which might be more trouble than it's worth.
On Aug 11, 2016 2:28 PM, "Tim Holy" <notifications@github.com> wrote:

> In test/dict.jl
> <https://github.com/JuliaLang/julia/pull/17968#discussion_r74476669>:
>
> > @@ -388,6 +388,11 @@ d = Dict('a'=>1, 'b'=>1, 'c'=> 3)
> >  @test [d[k] for k in keys(d)] == [d[k] for k in eachindex(d)] ==
> >        [v for (k, v) in d] == [d[x[1]] for (i, x) in enumerate(d)]
> >
> > +# generators, similar
> > +d = Dict(:a=>"a")
> > +@test @inferred(map(identity, d)) == d
> > +@test @inferred(map(p->p.first=>p.second[1], d)) == Dict(:a=>'a')
> > +@test @inferred(map(p->p.second, d)) == Any["a"]
>
> Sounds good. Should this be a deprecation or an outright error? In other
> words, would you be OK with moving the array-creating version to
> deprecated.jl?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/JuliaLang/julia/pull/17968/files/3cfc74d47ddf7ffc9e9db32e3057ce648438fe1c#r74476669>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAtcbAYxqgPJaOFl02SP9tqPgBnZe1_Eks5qe2nPgaJpZM4JiQcr>
> .
>